org.eclipse.m2e.pde.connector/src/org/eclipse/m2e/pde/connector/TychoPackagingsConfigurator.java

```diff
@@ -0,0 +1,84 @@
+ *******************************************************************************/
+package org.eclipse.m2e.pde.connector;
+
+import java.util.List;
+
+import org.apache.maven.plugin.MojoExecution;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ProjectScope;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.m2e.core.project.configurator.AbstractProjectConfigurator;
+import org.eclipse.m2e.core.project.configurator.ProjectConfigurationRequest;
+import org.eclipse.pde.internal.core.natures.PDE;
+import org.eclipse.pde.internal.core.util.CoreUtility;
+
+public class TychoPackagingsConfigurator extends AbstractProjectConfigurator {
+
+    private static final String TYCHO_DS_PLUGIN_GROUP_ID = "org.eclipse.tycho";
+    private static final String TYCHO_DS_PLUGIN_ARTIFACT_ID = "tycho-ds-plugin";
+    private static final String GOAL_DECLARATIVE_SERVICES = "declarative-services";
+
+    @Override
+    public void configure(ProjectConfigurationRequest request, IProgressMonitor monitor) throws CoreException {
+        MavenProject mavenProject = request.mavenProject();
+        IProject project = request.mavenProjectFacade().getProject();
+
+        String packaging = mavenProject.getPackaging();
+        if ("eclipse-plugin".equals(packaging) || "eclipse-test-plugin".equals(packaging)) {
+            PDEProjectHelper.configurePDEBundleProject(project, mavenProject, monitor);
+            applyDsConfiguration(request, monitor);
+        } else if ("eclipse-feature".equals(packaging)) {
+            // see org.eclipse.pde.internal.ui.wizards.feature.AbstractCreateFeatureOperation
+            if (!project.hasNature(PDE.FEATURE_NATURE)) {
+                CoreUtility.addNatureToProject(project, PDE.FEATURE_NATURE, monitor);
+            }
+        } else if ("eclipse-repository".equals(packaging)) {
+            // see org.eclipse.pde.internal.ui.wizards.site.NewSiteProjectCreationOperation
+            if (!project.hasNature(PDE.SITE_NATURE)) {
+                CoreUtility.addNatureToProject(project, PDE.SITE_NATURE, monitor);
+            }
+        }
+    }
+
+    void applyDsConfiguration(ProjectConfigurationRequest request, IProgressMonitor monitor) throws CoreException {
+        List<MojoExecution> mojoExecutions = getTychoDsPluginMojoExecutions(request, monitor);
+        for (MojoExecution mojoExecution : mojoExecutions) {
+            // apply PDE configuration for DS
+            IEclipsePreferences prefs = new ProjectScope(request.mavenProjectFacade().getProject())
+                    .getNode(org.eclipse.pde.ds.internal.annotations.Activator.PLUGIN_ID);
+            Xpp3Dom dom = mojoExecution.getConfiguration();
+            Xpp3Dom dsEnabled = dom.getChild("tycho.ds.enabled");
```

Comment:
This seems to be the user property but you want the xml element name (`enabled`), same below with tycho.ds.version (`version`)...

Also I think we must check for the path and should set (if not already configured)

```
generateBundleActivationPolicyLazy=true
validationErrorLevel=error
validationErrorLevel.missingImplicitUnbindMethod=error
```

Reply:
good catch regarding XML element names.
For the default values, I leave that up to https://github.com/eclipse-pde/eclipse.pde/blob/8312046eb6c566df0846f4125c2a5d41dcf0d0a3/ds/org.eclipse.pde.ds.annotations/src/org/eclipse/pde/ds/internal/annotations/PreferenceInitializer.java#L23. Path should in no case be set to the value from this plugin as this plugins writes to /target/OSGI-INF and PDE writes to /OSGI-INF.

Reply:
> writes to /target/OSGI-INF and PDE writes to /OSGI-INF.

Still the config option is always "OSGI-INF" ...

Reply:
Not necessarily if someone gives an absolute path. This is not a read-only param and may be changed.

Reply:
> Not necessarily if someone gives an absolute path. This is not a read-only param and may be changed.

Then this is a configuration error. An absolute path is not useful here.

Reply:
Which path do you consider useful apart from the default "OSGI-INF".

Reply:
It doesn't matter what I consider, per the specification you can place it in any folder, (e.g. one named 'components'), and you can configure this in PDE, there is an option in Tycho, so why not simply apply it as the other options as well? :-)